src/stages/stage6/draw.c

```diff
@@ -0,0 +1,328 @@
+/*
+ * This software is licensed under the terms of the MIT License.
+ * See COPYING for further information.
+ * ---
+ * Copyright (c) 2011-2019, Lukas Weber <laochailan@web.de>.
+ * Copyright (c) 2012-2019, Andrei Alexeyev <akari@taisei-project.org>.
+ */
+
+#include "taisei.h"
+
+#include "draw.h"
+
+#include "global.h"
+
+static Stage6DrawData *stage6_draw_data;
+
+Stage6DrawData *stage6_get_draw_data(void) {
+	return NOT_NULL(stage6_draw_data);
+}
+
+void stage6_drawsys_init(void) {
+	stage6_draw_data = calloc(1, sizeof(*stage6_draw_data));
+	stage3d_init(&stage_3d_context, 16);
+
+	// TODO: make this background slightly less horribly inefficient
+
+	stage3d_init(&stage_3d_context, 128);
+	stage6_draw_data->fall_over.frames = 0;
```

Comment:
Just tested this. Everything is fine, except there's now a memory leak because `stage3d_init` is called twice here.